http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcRequestAuthenticator.java

```diff
@@ -416,4 +422,26 @@ private static boolean hasScope(String scopeParam, String targetScope) {
         }
         return false;
     }
+
+    private String combineAndReorderScopes(Set<String> allScopes, String paramValue) {
+        StringBuilder combinedScopes = new StringBuilder();
+        addScope(paramValue, allScopes);
+
+        if (allScopes.contains(OIDC_SCOPE)) { //some OpenID providers require openid scope to be added in the beginning
+            allScopes.remove(OIDC_SCOPE);
+        }
+        combinedScopes.append(OIDC_SCOPE);
+        for (String scope : allScopes) {
+            if (scope != null && !scope.isEmpty()) {
```

Comment:
Can the scope be null here?

Reply:
I have removed it. 
I was thinking if someone adds more than one space in between, then one of the scopes could be null/empty. But I think that is handled already